### DISCUSSION TITLE
[Question][Hw2] What does "cmp pred v1 v2" do? And about "Llvm.Icmp.t".

### DISCUSSION THREAD

**Discussion by sjoon2455** [prosyslab-classroom/cs524-program-analysis#420] [Q&A]
Hi, I want to ask you about the functionality of ```cmp pred v1 v2```.
My understanding is that ```pred``` would be evaluated to either ```true``` or ```false```, and the whole function call will return ```v1``` or ```v2``` accordingly.
Or, ```pred``` is sth like ```>``` and it is used to compare ```v1``` and ```v2```, which eventually returns ```true``` or ```false```.
The latter seems to be correct according to my search on Google, but the term ```pred``` confuses me in that predicate means "something that is evaluated to either true or false".  Which one is correct?
Also regardless of the above, I wonder if any kind of document or specification regarding such explanation is given. 

One more question! I'm having difficulty with ```Llvm.Icmp.t``` which is seemingly a type of ```pred```.
I've read https://github.com/prosyslab-classroom/llvm-primer/blob/master/data-structure.md and https://llvm.moe/ocaml-3.6/type_Llvm.html, but couldn't really find out the way out. Any hint to evaluate ```pred``` to a type ```t```?

Thanks!

**Comment by hyunsukimsokcho** [comment]
1. The latter is the right one.
2. Try to use pattern matching.

**Comment by hyunsukimsokcho** [comment]
closing.. please feel free to re-open if needed.
